libpromises/cf3lex.l

```diff
@@ -298,6 +377,22 @@ promise_type   [a-zA-Z_]+:
                           return yytext[0];
                       }
 
+<if_ignore_state><<EOF>>     {
+                               if (P.if_depth > 0)
+                               {
+                                 yyerror("EOF seen while @if was waiting for @endif");
+                                 exit(1);
+                               }
+                             }
+
+<<EOF>>     {
+              if (P.if_depth > 0)
+              {
+                yyerror("EOF seen while @if was waiting for @endif");
+                exit(1);
+              }
+              return 0; // loops forever without this
+            }
```

Comment:
Without the explicit `<<EOF>>` section, the lexer accepts cases like `/00_basics/macros/if_eof_without_endif2.x.cf`.

With the explicit `<<EOF>>` section but without an explicit `return 0`, the lexer never finishes parsing any of the tests.

I don't want to assume the `P.if_depth` logic is perfect, so I'd rather leave the check in.